File: src/latihanoop/BukuDipinjam.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package latihanoop;

import java.util.ArrayList;

/**
 *
 * @author farha
 */
public class BukuDipinjam extends Buku {
    private ArrayList<Anggota> peminjam;
    
    public BukuDipinjam(int ID, String Penulis, String Judul, int Stok) {
        super(ID, Penulis, Judul, Stok);
        this.peminjam = new ArrayList<>(); // Inisialisasi daftar peminjam
    }

    public ArrayList<Anggota> getPeminjam() {
        return peminjam;
    }

    public void setPeminjam(ArrayList<Anggota> peminjam) {
        this.peminjam = peminjam;
    }
    
    @Override
    public String printInfo() {
        String str = "";
        for (Anggota anggota : peminjam) {
            str += anggota.getNama() + ", ";
        }
        return "ID Buku\t:" + getID() + "\n" + 
                "Judul Buku\t:" + getJudul() + "\n" +
                "Penulis\t:" + getPenulis() + "\n" +
                "Peminjam\t: " + str + "\n";
    }
}

File: app/src/main/java/lsp/project/borrowshelf/data/BukuDipinjam.kt
package lsp.project.borrowshelf.data

class BukuDipinjam(
  id: Int,
  penulis: String,
  judul: String,
  stok: Int
) : Buku(id, penulis, judul, stok) {

  var peminjam: ArrayList<Anggota> = ArrayList()

  override fun printInfo(): String {
    val str = peminjam.joinToString(", ") { it.nama }
    return """
            ID Buku	    : $id
            Judul Buku	: $judul
            Penulis	    : $penulis
            Peminjam	  : $str
        """.trimIndent()
  }
}
